modules/gdscript/gdscript_warning.cpp

```diff
@@ -113,13 +113,9 @@ String GDScriptWarning::get_message() const {
 			return R"(The "@static_unload" annotation is redundant because the file does not have a class with static variables.)";
 		case REDUNDANT_AWAIT:
 			return R"("await" keyword not needed in this case, because the expression isn't a coroutine nor a signal.)";
-		case REDUNDANT_FOR_VARIABLE_TYPE:
+		case FOR_VARIABLE_EXPLICIT_SUPERTYPE:
 			CHECK_SYMBOLS(3);
-			if (symbols[1] == symbols[2]) {
-				return vformat(R"(The for loop iterator "%s" already has inferred type "%s", the specified type is redundant.)", symbols[0], symbols[1]);
-			} else {
-				return vformat(R"(The for loop iterator "%s" has inferred type "%s" but its supertype "%s" is specified.)", symbols[0], symbols[1], symbols[2]);
-			}
+			return vformat(R"(The for loop iterator "%s" has inferred type "%s" but its supertype "%s" is specified.)", symbols[0], symbols[1], symbols[2]);
```

Comment:
```suggestion
			return vformat(R"(The for loop iterator "%s" has inferred type "%s" but its strict supertype "%s" is specified.)", symbols[0], symbols[1], symbols[2]);
```

I'm not sure if we should clarify this here, probably yes.